tests/test_dupefilters.py

```diff
@@ -78,20 +78,24 @@ def test_filter(self):
     def test_dupefilter_path(self):
         r1 = Request('http://scrapytest.org/1')
         r2 = Request('http://scrapytest.org/2')
+        r3 = Request('http://scrapytest.org/3')
 
         path = tempfile.mkdtemp()
         try:
             df = RFPDupeFilter(path)
             df.open()
             assert not df.request_seen(r1)
             assert df.request_seen(r1)
+            assert not df.request_seen(r2)
+            assert df.request_seen(r2)
             df.close('finished')
 
             df2 = RFPDupeFilter(path)
             df2.open()
             assert df2.request_seen(r1)
-            assert not df2.request_seen(r2)
             assert df2.request_seen(r2)
+            assert not df2.request_seen(r3)
+            assert df2.request_seen(r3)
```

Comment:
I added these so I could be certain the disk-based cache correctly restores several with different newlines/delimiter on disk.